src/annotations.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+#ifndef TIMESCALEDB_ANNOTATIONS_H
+#define TIMESCALEDB_ANNOTATIONS_H
+
+/* Fall-through annotation */
+#if defined(__clang__)
+#if (__clang_major__ >= 12) || (__clang_analyzer__)
+#define TS_FALLTHROUGH __attribute__((fallthrough))
+#else
+#define TS_FALLTHROUGH /* FALLTHROUGH */
```

Comment:
One thing that worries me here is that both comments and macros symbols are handled by the preprocessor. Have you tested this with a compiler that uses the `/* FALLTHROUGH */` syntax? I think that the standard builds-on-push do that but can you please verify before pushing.

Reply:
We used the `/* FALLTHROUGH */ syntax prior to this PR.  But clang seems to have dropped support for it in favor of the GCC-compatible attribute. AFAIK, the CI still runs a pre 12 clang, so should still use `/* FALLTHROUGH */`.